File: evals/refs/pop.zig
const std = @import("std");
pub fn main(init: std.process.Init) !void {
    const arena = init.arena.allocator();
    var in_buf: [4096]u8 = undefined;
    var fr = std.Io.File.stdin().reader(init.io, &in_buf);
    const r = &fr.interface;
    var data: std.ArrayList(u8) = .empty;
    try r.appendRemainingUnlimited(arena, &data);
    var out_buf: [4096]u8 = undefined;
    var fw = std.Io.File.stdout().writer(init.io, &out_buf);
    const w = &fw.interface;
    var it = std.mem.tokenizeAny(u8, data.items, " \n\r\t");
    const t = try std.fmt.parseInt(usize, it.next().?, 10);
    var c: usize = 0;
    while (c < t) : (c += 1) {
        var n = try std.fmt.parseInt(u64, it.next().?, 10);
        var count: u32 = 0;
        while (n > 0) {
            n &= n - 1;
            count += 1;
        }
        try w.print("{d}\n", .{count});
    }
    try w.flush();
}

File: evals/refs/pop.rs
use std::io::{self, Read, Write, BufWriter};

fn main() {
    let mut s = String::new();
    io::stdin().read_to_string(&mut s).unwrap();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut it = s.split_ascii_whitespace();
    let t: usize = it.next().unwrap().parse().unwrap();
    for _ in 0..t {
        let mut n: u64 = it.next().unwrap().parse().unwrap();
        let mut count: u32 = 0;
        while n > 0 {
            n &= n - 1;
            count += 1;
        }
        writeln!(out, "{}", count).unwrap();
    }
}
